lib/Newsletter/Shortcodes/Shortcodes.php

```diff
@@ -80,19 +80,60 @@ public function extract($content, $categories = false) {
       false;
   }
 
+  /**
+   * Parse a MailPoet-style shortcode.
+   * The syntax is [category:action | argument:argument_value], it can have a single argument.
+   */
   public function match($shortcode) {
     preg_match(
       '/\[(?P<category>\w+)?:(?P<action>\w+)(?:.*?\|.*?(?P<argument>\w+):(?P<argument_value>.*?))?\]/',
       $shortcode,
       $match
     );
+    // If argument exists, copy it to the arguments array
+    if (!empty($match['argument'])) {
+      $match['arguments'] = [$match['argument'] => isset($match['argument_value']) ? $match['argument_value'] : ''];
+    }
     return $match;
   }
 
+  /**
+   * Parse a WordPress-style shortcode.
+   * The syntax is [category:action arg1="value1" arg2="value2"], it can have multiple arguments.
+   */
+  public function matchWPShortcode($shortcode) {
+    $atts = WPFunctions::get()->shortcodeParseAtts(trim($shortcode, '[]/'));
```

Comment:
We are already using DI in this class, you don't need to use `WPFunctions::get()`. You can inject the functions class directly.

Reply:
Fixed.